Failure building Athenz docker image

Following the steps in docker/Readme.md didn't work.

1. The instructions suggest using an old tag:
`P.S. the latest code may cause docker build to fail, please use older version by specifying the tag version (< v1.9.27) or post an issue
 make build TAG=v1.9.27`

But using an older tag fails because the docker/util/rdl-athenz-go-model directory didn't exist until a month ago and the images require it.
Probably just need to remove this instruction.

2. When running "make build" without tag the build still fails on the following:
`executor failed running [/bin/sh -c mvn install -pl core/zms -pl core/zts -pl libs/java/auth_core -pl libs/java/client_common -pl libs/java/server_common -pl libs/java/instance_provider -pl libs/java/cert_refresher -pl clients/java/zms -pl servers/zms -pl clients/java/zts/core -pl servers/zts -pl containers/jetty -pl assembly/zms -pl assembly/zts -Dmaven.test.skip=true -Djacoco.skip=true]: exit code: 1
make: *** [build] Error 1`
This happens because it looks for the directories under the "docker" directory instead of the main athenz directory

I think it's related to the changes in https://github.com/AthenZ/athenz/pull/1373.
When trying to build from a previous commit, I pass this stage but get an error on creating tar archive:
` => [5/9] RUN mkdir -p /tmp/zms     && mkdir -p /tmp/zts     && mkdir -p /opt/athenz/zms     && mkdir -p /opt/athenz/zts                                                                                 0.6s
 => ERROR [6/9] RUN tar xvzf ./assembly/zms/target/athenz-zms*-bin.tar.gz -C /tmp/zms     && mv /tmp/zms/athenz-zms-*/* /tmp/zms/     && mv /tmp/zms/etc /opt/athenz/zms/etc     && mv /tmp/zms/conf /o  1.0s
------`

--- a/.gitignore
+++ b/.gitignore
@@ -113,6 +113,7 @@ rdl/rdl-gen-athenz-server/rdl-gen-athenz-server
 rdl/rdl-gen-athenz-go-client/rdl-gen-athenz-go-client
 rdl/rdl-gen-athenz-go-model/rdl-gen-athenz-go-model
 rdl/rdl-gen-athenz-java-model/rdl-gen-athenz-java-model
+rdl/rdl-gen-athenz-java-client/rdl-gen-athenz-java-client
 libs/go/athenzutils/pkg/
 **test-output**
 clients/nodejs/zpe/package-lock.json

--- a/docker/Makefile
+++ b/docker/Makefile
@@ -31,6 +31,7 @@ else
 endif
 	DOCKER_BUILDKIT=1 docker build $(BUILD_ARG) -t rdl-athenz-server -f ./util/rdl-athenz-server/Dockerfile ../rdl/rdl-gen-athenz-server
 	DOCKER_BUILDKIT=1 docker build $(BUILD_ARG) -t rdl-athenz-java-model -f ./util/rdl-athenz-java-model/Dockerfile ../rdl/rdl-gen-athenz-java-model
+	DOCKER_BUILDKIT=1 docker build $(BUILD_ARG) -t rdl-athenz-java-client -f ./util/rdl-athenz-java-client/Dockerfile ../rdl/rdl-gen-athenz-java-client
 	DOCKER_BUILDKIT=1 docker build $(BUILD_ARG) -t rdl-athenz-go-model -f ./util/rdl-athenz-go-model/Dockerfile ../rdl/rdl-gen-athenz-go-model
 	DOCKER_BUILDKIT=1 docker build $(BUILD_ARG) -t athenz-mvn-base -f ./util/athenz-mvn-base/Dockerfile ../
 	DOCKER_BUILDKIT=1 docker build $(BUILD_ARG) -t athenz-builder -f ./util/athenz-builder/Dockerfile ../

--- a/docker/util/athenz-mvn-base/Dockerfile
+++ b/docker/util/athenz-mvn-base/Dockerfile
@@ -6,6 +6,8 @@ FROM rdl-athenz-java-model AS rdl-java-model
 
 FROM rdl-athenz-go-model AS rdl-go-model
 
+FROM rdl-athenz-java-client AS rdl-java-client
+
 FROM maven:3.6-alpine AS mvn
 # date -u +'%Y-%m-%dT%H:%M:%SZ'
 ARG BUILD_DATE
@@ -27,6 +29,7 @@ ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
 COPY --from=rdl /usr/bin/rdl /usr/bin/rdl
 COPY --from=rdl /usr/bin/rdl-gen-athenz-server /usr/bin/rdl-gen-athenz-server
 COPY --from=rdl-java-model /usr/bin/rdl-gen-athenz-java-model /usr/bin/rdl-gen-athenz-java-model
+COPY --from=rdl-java-client /usr/bin/rdl-gen-athenz-java-client /usr/bin/rdl-gen-athenz-java-client
 COPY --from=rdl-go-model /usr/bin/rdl-gen-athenz-go-model /usr/bin/rdl-gen-athenz-go-model
 
 # copy pom.xml
@@ -35,6 +38,7 @@ COPY pom.xml .
 # cat pom.xml | grep "<module>.*</module>" | awk -F'[><]' '{ printf "COPY ./%s/pom.xml ./%s/pom.xml\n", $3, $3 }'
 COPY ./rdl/rdl-gen-athenz-server/pom.xml ./rdl/rdl-gen-athenz-server/pom.xml
 COPY ./rdl/rdl-gen-athenz-java-model/pom.xml ./rdl/rdl-gen-athenz-java-model/pom.xml
+COPY ./rdl/rdl-gen-athenz-java-client/pom.xml ./rdl/rdl-gen-athenz-java-client/pom.xml
 COPY ./rdl/rdl-gen-athenz-go-model/pom.xml ./rdl/rdl-gen-athenz-go-model/pom.xml
 COPY ./rdl/rdl-gen-athenz-go-client/pom.xml ./rdl/rdl-gen-athenz-go-client/pom.xml
 COPY ./core/zms/pom.xml ./core/zms/pom.xml

--- a/docker/util/rdl-athenz-java-client/Dockerfile
+++ b/docker/util/rdl-athenz-java-client/Dockerfile
@@ -1,0 +1,45 @@
+FROM golang:1.14-alpine AS builder
+
+ENV APP_NAME rdl-gen-athenz-java-client
+
+RUN set -eux \
+    && apk --no-cache add --virtual build-dependencies cmake g++ make unzip curl upx git
+
+WORKDIR ${GOPATH}/src/github.com/AthenZ/athenz/rdl/rdl-gen-athenz-java-client
+
+RUN go get -u github.com/ardielle/ardielle-go/... \
+    && go get -u github.com/ardielle/ardielle-tools/... \
+    && mv ${GOPATH}/bin/rdl /usr/bin/rdl
+
+COPY . .
+
+RUN CGO_ENABLED=1 \
+    CGO_CXXFLAGS="-g -Ofast -march=native" \
+    CGO_FFLAGS="-g -Ofast -march=native" \
+    CGO_LDFLAGS="-g -Ofast -march=native" \
+    GOOS=$(go env GOOS) \
+    GOARCH=$(go env GOARCH) \
+    go build --ldflags '-s -w -linkmode "external" -extldflags "-static -fPIC -m64 -pthread -std=c++11 -lstdc++"' -a -tags "cgo netgo" -installsuffix "cgo netgo" -o "${APP_NAME}" \
+    && upx -9 -o "/usr/bin/${APP_NAME}" "${APP_NAME}"
+
+RUN apk del build-dependencies --purge \
+    && rm -rf "${GOPATH}"
+
+# Start From Scratch For Running Environment
+FROM scratch
+# date -u +'%Y-%m-%dT%H:%M:%SZ'
+ARG BUILD_DATE
+# git rev-parse --short HEAD
+ARG VCS_REF
+
+LABEL org.label-schema.schema-version="1.0"
+LABEL org.label-schema.build-date=$BUILD_DATE
+LABEL org.label-schema.name="rdl-athenz-java-client"
+LABEL org.label-schema.description="base image of RDL"
+LABEL org.label-schema.url="https://www.athenz.io/"
+LABEL org.label-schema.vcs-url="https://github.com/AthenZ/athenz"
+LABEL org.label-schema.vcs-ref=$VCS_REF
+
+ENV APP_NAME rdl-gen-athenz-java-client
+
+COPY --from=builder /usr/bin/${APP_NAME} /usr/bin/${APP_NAME}

